src/target/adiv5_swd.c

```diff
@@ -291,18 +291,21 @@ void adiv5_swd_multidrop_scan(adiv5_debug_port_s *const dp, const uint32_t targe
 		 */
 
 		/* Line reset sequence */
+		uint32_t target = instance << ADIV5_DP_TARGETSEL_TINSTANCE_OFFSET |
+				(targetid & (ADIV5_DP_TARGETID_TDESIGNER_MASK | ADIV5_DP_TARGETID_TPARTNO_MASK)) | 1U;
 		swd_line_reset_sequence(true);
 		dp->fault = 0;
-
-		/* Select the instance */
-		dp->dp_low_write(ADIV5_DP_TARGETSEL,
-			instance << ADIV5_DP_TARGETSEL_TINSTANCE_OFFSET |
-				(targetid & (ADIV5_DP_TARGETID_TDESIGNER_MASK | ADIV5_DP_TARGETID_TPARTNO_MASK)) | 1U);
-
-		/* Read DPIDR */
-		if (adiv5_dp_read_dpidr(dp) == 0)
-			/* No DP here, next instance */
-			continue;
+		swd_proc.seq_out(make_packet_request(ADIV5_LOW_WRITE, ADIV5_DP_TARGETSEL), 8U);
+		swd_proc.seq_in(3);
+		swd_proc.seq_out_parity(target, 32U);
+		swd_proc.seq_out(make_packet_request(ADIV5_LOW_READ, ADIV5_DP_DPIDR), 8U);
+		uint32_t ack = swd_proc.seq_in(3);
+		/* DPIDR may not issue WAIT! */
+		if ((ack & 0x7) !=SWDP_ACK_OK)
+			continue; /* No DP here, next instance */
+		uint32_t dpidr;
+		if (swd_proc.seq_in_parity(&dpidr, 32U) || (dpidr == 0))
```

Comment:
The inner parens here are extraneous, please drop them so we get the GCC warning about accidental assignment. Please suffix all integers here with U as they're unsigned constants, so make them unsigned.

As much as we admire this attempt to fix the "SWD access resulted in no response" outputs on scan, this whole change feels very misguided, and especially as this will increase Flash footprint and is way harder to read and reason about. It's not a problem that there's a line of "SWD access resulted in no response" that gets emitted as it's being truthful about what's going on under the hood. Unless you can strongly justify this change *and include proper comments in the code about what it's doing*, we're going to ask you to revert this.